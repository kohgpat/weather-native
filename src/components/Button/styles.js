import styled from "styled-components/native";

export const Button = styled.Button`
  font-size: 16px;
  font-weight: bold;
  padding: 8px 16px;
  height: 40px;
  width: 100%;
  border: 0;
  background-color: #4FADDF;
  color: #FFF;
`;
