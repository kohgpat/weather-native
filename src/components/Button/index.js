import React from "react";
import * as styles from "./styles";

const Button = ({ title, ...restProps }) => (
  <styles.Button {...restProps}>
    <styles.ButtonTitle>{title}</styles.ButtonTitle>
  </styles.Button>
);

export default Button;
