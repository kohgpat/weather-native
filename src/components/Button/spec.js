/*global it, expect*/
import React from "react";
import Button from "./index";
import renderer from 'react-test-renderer';

it("renders", () => {
  const button = renderer.create(<Button>Button</Button>).toJSON();
  expect(button).toMatchSnapshot();
});
