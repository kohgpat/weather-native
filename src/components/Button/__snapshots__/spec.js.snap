// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#4FADDF",
      "height": 40,
      "justifyContent": "center",
      "opacity": 1,
      "width": "100%",
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#fff",
          "fontFamily": "Roboto-Condensed-Bold",
          "fontSize": 16,
        },
        undefined,
      ]
    }
  />
</View>
`;
